api-compose-object.go

```diff
@@ -114,7 +114,7 @@ func (opts CopyDestOptions) Marshal(header http.Header) {
 	}
 
 	if opts.Encryption != nil {
-		opts.Encryption.Marshal(header)
+		opts.Encryption.MarshalCOPY2(header)
```

Comment:
it will be more readable to rename MarshalCOPY -> MarshalCOPYSrc and MarshalCOPY2 -> MarshalCOPYDest

Reply:
My idea was `MarshalCOPY2` reads like `MarshalCOPY to` referring to `copy from` / `copy to`.
However, I agree that it may be too confusing when reading code using the interface...

Will try to find a better name (that also is a idiomatic Go name)